addons/minio/templates/cpmd.yaml

```diff
@@ -0,0 +1,166 @@
+{{ $scheme := .Values.tls.enabled | ternary "https" "http" }}
+
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: ComponentDefinition
+metadata:
+  name: minio
+  labels:
+    {{- include "minio.labels" . | nindent 4 }}
+spec:
+  provider: kubeblocks
+  description: Minio is a High Performance Object Storage.
+  serviceKind: minio
+  serviceVersion: {{ .Chart.Version }}
+
+  systemAccounts:
+    - name: root
+      initAccount: true
+      passwordGenerationPolicy:
+        length: 16
+        numDigits: 8
+        numSymbols: 2
+        letterCase: MixedCases
+
+  vars:
+    - name: MINIO_ROOT_USER
+      valueFrom:
+        credentialVarRef:
+          name: root
+          optional: false
+          username: Required
+    - name: MINIO_ROOT_PASSWORD
+      valueFrom:
+        credentialVarRef:
+          name: root
+          optional: false
+          password: Required
+
+  lifecycleActions:
+    roleProbe:
+      builtinHandler: custom
+      customHandler:
+        exec:
+          command:
+            - /bin/bash
+            - -c
+            - |
+              mc config host add minio http://127.0.0.1:9000 $MINIO_ROOT_USER $MINIO_ROOT_PASSWORD
+
+  services:
+  - name: default
+    spec:
+      ports:
+        - name: api
+          port: 9000
+          targetPort: api
+        - name: console
+          port: 9001
+          targetPort: console
+
+  configs:
+    - name: minio-configuration
+      templateRef: minio-configuration
+      namespace: {{ .Release.Namespace }}
+      volumeName: minio-config
+      defaultMode: 0777
+
+  runtime:
+    initContainers:
+    - command:
+      - /bin/sh
+      - -ce
+      - |
+        name="$KB_CLUSTER_COMP_NAME"-minio-configuration
+        namespace="$KB_NAMESPACE"
+        key="MINIO_REPLICAS_HISTORY"
+        cur=$(kubectl get configmaps "$name" -n "$namespace" -o jsonpath="{.data.$key}")
+        cur=$(echo "$cur" | tr -d '[]')
+        if [[ -z "$cur" ]]; then
+          new="[$KB_COMP_REPLICAS]"
+        else
+          IFS=',' set -- $cur
+          array="$@"
+          last=$(echo $array | awk '{print $NF}')
+          if [[ "$last" == "$KB_COMP_REPLICAS" ]]; then
+            new="[$cur]"
+          else
+            new="[$cur,$KB_COMP_REPLICAS]"
+          fi
+        fi
+        kubectl patch configmap "$name" -n "$namespace" --type strategic -p "{\"data\":{\"$key\":\"$new\"}}"
```

Comment:
It is very hacky to manipulate a configmap in init container, if the Pod is restricted by RBAC to write against a configmap, this command will fail. Try to use vars and templates to render the configmap.

Reply:
For MinIO expansion, I need to concatenate commands using the history of replicas. Therefore, I've recorded this information in ConfigMaps. And now, the service account (SA) created by default for the cluster already has the necessary permissions. However, it would be better if there were other methods to record this information.